backend/pkg/api/data_access/vdb_blocks.go

```diff
@@ -32,319 +41,272 @@ func (d *DataAccessService) GetValidatorDashboardBlocks(ctx context.Context, das
 		}
 	}
 
-	// regexes taken from api handler common.go
 	searchPubkey := regexp.MustCompile(`^0x[0-9a-fA-F]{96}$`).MatchString(search)
 	searchGroup := regexp.MustCompile(`^[a-zA-Z0-9_\-.\ ]+$`).MatchString(search)
 	searchIndex := regexp.MustCompile(`^[0-9]+$`).MatchString(search)
 
-	validatorMap := make(map[t.VDBValidator]bool)
-	params := []interface{}{}
-	filteredValidatorsQuery := ""
-	validatorMapping, err := d.services.GetCurrentValidatorMapping()
-	if err != nil {
-		return nil, nil, err
+	validators := goqu.T("users_val_dashboards_validators").As("validators")
+	blocks := goqu.T("blocks")
+	groups := goqu.T("groups")
+
+	type validatorGroup struct {
+		Validator t.VDBValidator `db:"validator_index"`
+		Group     uint64         `db:"group_id"`
 	}
 
-	// determine validators of interest first
-	if dashboardId.Validators == nil {
-		// could also optimize this for the average and/or the whale case; will go with some middle-ground, needs testing
-		// (query validators twice: once without search applied (fast) to pre-filter scheduled proposals (which are sent to db, want to minimize),
-		// again for blocks query with search applied to not having to send potentially huge validator-list)
-		startTime := time.Now()
-		valis, err := d.getDashboardValidators(ctx, dashboardId, nil)
-		log.Debugf("=== getting validators took %s", time.Since(startTime))
-		if err != nil {
-			return nil, nil, err
-		}
-		for _, v := range valis {
-			validatorMap[v] = true
-		}
+	// -------------------------------------
+	// Goqu Query to determine validators filtered by search
+	var filteredValidatorsDs *goqu.SelectDataset
+	var filteredValidators []validatorGroup
 
-		// create a subquery to get the (potentially filtered) validators and their groups for later
-		params = append(params, dashboardId.Id)
-		selectStr := `SELECT validator_index, group_id `
-		from := `FROM users_val_dashboards_validators validators `
-		where := `WHERE validators.dashboard_id = $1`
-		extraConds := make([]string, 0, 3)
+	filteredValidatorsDs = goqu.Dialect("postgres").
+		Select(
+			"validator_index",
+		)
+	if dashboardId.Validators == nil {
+		filteredValidatorsDs = filteredValidatorsDs.
+			From(validators).
+			Where(validators.Col("dashboard_id").Eq(dashboardId.Id))
+		// apply search filters
 		if searchIndex {
-			params = append(params, search)
-			extraConds = append(extraConds, fmt.Sprintf(`validator_index = $%d`, len(params)))
+			filteredValidatorsDs = filteredValidatorsDs.Where(validators.Col("validator_index").Eq(search))
```

Comment:
**issue:** Constructing the search like this doesn't work because the WHERE all get joined by an AND.
So if you search for an index or pubkey the format also qualifies a group name.
Then you only find results where both the index/pubkey and the group name work for the filter.